_episodes/06-exploring_history.md

```diff
@@ -12,6 +12,605 @@ objectives:
++The two moons may be a problem for Wolfman
++But the Mummy will appreciate the lack of humidity
++An ill-considered change
+```
+
+We could also use `git show` which shows us what changes we made at an older commit as
+well as the commit message, rather than the *differences* between a commit and our
+working directory that we see by using `git diff`.
+
+```bash
+$ git show HEAD~3 mars.txt
+```
+
+```output
+commit f22b25e3233b4645dabd0d81e651fe074bd8e73b
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Thu Aug 22 09:51:46 2013 -0400
+
+    Start notes on Mars as a base
+
+diff --git a/mars.txt b/mars.txt
+new file mode 100644
+index 0000000..df0654a
+--- /dev/null
++++ b/mars.txt
+@@ -0,0 +1 @@
++Cold and dry, but everything is my favorite color
+```
+
+In this way,
+we can build up a chain of commits.
+The most recent end of the chain is referred to as `HEAD`;
+we can refer to previous commits using the `~` notation,
+so `HEAD~1`
+means "the previous commit",
+while `HEAD~123` goes back 123 commits from where we are now.
+
+We can also refer to commits using
+those long strings of digits and letters
+that `git log` displays.
+These are unique IDs for the changes,
+and "unique" really does mean unique:
+every change to any set of files on any computer
+has a unique 40-character identifier.
+Our first commit was given the ID
+`f22b25e3233b4645dabd0d81e651fe074bd8e73b`,
+so let's try this:
+
+```bash
+$ git diff f22b25e3233b4645dabd0d81e651fe074bd8e73b mars.txt
+```
+
+```output
+diff --git a/mars.txt b/mars.txt
+index df0654a..93a3e13 100644
+--- a/mars.txt
++++ b/mars.txt
+@@ -1 +1,4 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
++But the Mummy will appreciate the lack of humidity
++An ill-considered change
+```
+
+That's the right answer,
+but typing out random 40-character strings is annoying,
+so Git lets us use just the first few characters (typically seven for normal size projects):
+
+```bash
+$ git diff f22b25e mars.txt
+```
+
+```output
+diff --git a/mars.txt b/mars.txt
+index df0654a..93a3e13 100644
+--- a/mars.txt
++++ b/mars.txt
+@@ -1 +1,4 @@
+ Cold and dry, but everything is my favorite color
++The two moons may be a problem for Wolfman
++But the Mummy will appreciate the lack of humidity
++An ill-considered change
+```
+
+All right! So
+we can save changes to files and see what we've changed. Now, how
+can we restore older versions of things?
+Let's suppose we change our mind about the last update to
+`mars.txt` (the "ill-considered change").
+
+`git status` now tells us that the file has been changed,
+but those changes haven't been staged:
+
+```bash
+$ git status
+```
+
+```output
+On branch main
+Changes not staged for commit:
+  (use "git add <file>..." to update what will be committed)
+  (use "git checkout -- <file>..." to discard changes in working directory)
+
+    modified:   mars.txt
+
+no changes added to commit (use "git add" and/or "git commit -a")
+```
+
```

Comment:
I feel like all of these lessons would benefit from subheadings. For example, having a heading here showing the transition from `git diff` to `git checkout` would be nice for skimming and people perhaps finding where they left off. 

This would be a pretty big stylistic change and I wouldn't want this lesson to look different from others so perhaps we can discuss this as a broader change and go through at the end and make this update.

Reply:
I agree that is a larger stylistic change. This can be added as a larger issue for discussion since it would affect more than this review